habitat/sims/habitat_simulator/habitat_simulator.py

```diff
@@ -160,9 +164,11 @@ def get_observation(
 @registry.register_sensor
 class HabitatSimSemanticSensor(SemanticSensor):
     sim_sensor_type: habitat_sim.SensorType
+    sim_sensor_subtype: habitat_sim.SensorSubType
```

Comment:
I suppose, we could tie the SensorSubtype to the Sensor Class like this, but I was thinking about having it be a SensorConfig string that determined it (similar to how we do for UUIDs now). Any reason why not do it that way? It would give the user more flexability down the road (unless you expect SensorSubtypes to have entirely different config types).

Doing it like this though causes a quadratic explosion in the number of Sensor modalities -> sensor_subtypes though.

In other words, is there a good reason this cannot/should not be read from the config?

This is all fine for now since we only support PINHOLE, but will be annoying later when we supports RGBFisheye/DepthFisheye etc...

Reply:
Hi Aaron! I did this because this file goes through some defined configs and initializes the right sensorSpec. 

So far, only CameraSensorSpec exists, but I'm adding the structure for checks so that when Fisheye/Equirectangular sensors are implemented, we can follow the logic to initialize the right sensorSpec.

Reply:
Does Subtype infer how observation space will be defined? Does different Subtypes of sensors require separate classes wrappers in HabLab?

Reply:
Hi @mathfac , I think I don't understand enough about how observation space is defined in hablab vs habsim. What is the difference there and what should I be aware of?

Reply:
Exactly, I don't think they should require separate wrappers, they should be set by the config.